common/src/stack/report-system/command/report/system/fixtures/disk_partition_check.py

```diff
@@ -0,0 +1,191 @@
+import pytest
+import paramiko
+import socket
+import re
+import math
+from collections import namedtuple
+from itertools import groupby
+
+# Takes in a hostname and disk attributes, returns the lsblk data in a dictionary of disks with a nested
+# dictionary of partitions per disk which each has a named tuple of the name, mountpoint,
+# disk label, file system type, and disk size. Returns an empty dictionary if the host
+# cannot be ssh into or lsblk does not return any data
+def get_partition_data(hostname, attributes):
+
+	disks = {}
+
+	backend_ssh = paramiko.SSHClient()
+	backend_ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+	try:
+		backend_ssh.connect(hostname=f'{hostname}')
+
+	except (paramiko.SSHException, socket.error):
+		return {}
+
+	# Get backend host partition informaiton using lsblk
+	stdin, stdout, stderr = backend_ssh.exec_command(f'lsblk -r -n -b -o {",".join(attributes)}')
+
+	lsblk_data = stdout.readlines()
+
+	partition_attributes = namedtuple("partition", attributes)
+
+	# For each line in the lsblk output, make it into a tuple
+	# Then each partition into a dictionary keyed by it's disk
+	if lsblk_data:
+		for line in lsblk_data:
+			part_values = partition_attributes(*line.rstrip().split(' '))
+
+			# Convert bytes to megabytes if the size column was specified
+			if 'size' in attributes:
+				part_values = part_values._replace(size = int(part_values.size) / math.pow(2, 20))
+
+			# Get the name of the disk by taking the partition name without it's
+			# partition number
+			disk = ''.join(filter(lambda x: not x.isdigit(), part_values.name))
+
+			disks.setdefault(disk, {})[part_values.name] = part_values
+	else:
+		return {}
+
+	return disks
+
+
+def check_partition_size(partitions, curr_partition, expect_size, disk):
+	part_size = 0
+
+	if expect_size == '*':
+		return False
+
+	elif int(expect_size) == 0:
+		for partition, attributes in partitions.items():
+			if partition != disk and partition != curr_partition.name:
+				part_size += int(attributes.size)
+
+		remain_size = int(partitions[disk].size) - int(part_size)
+
+		if math.isclose(int(remain_size), int(curr_partition.size), abs_tol=100):
+			return True
+		else:
+			return False
+
+	elif not math.isclose( int(expect_size), int(curr_partition.size), abs_tol=100):
+		return False
+
+	else:
+		return True
+
+
+@pytest.fixture
+def verify_disk():
+	def partition_match_expected(expected_partitions, hostname, check_all_disks=False):
```

Comment:
This function is very large and I think would benefit from being subdivided into smaller functions that encapsulate blocks of the functionality, each with relevant docstrings and comments. For example, I see the `attribute_not_match` dictionary comprehension pattern repeated twice, which probably could move into a helper function.

I would also recommend defining this function in the module itself rather than inside the fixture function. This function can simply be returned from the fixture without needing to define it inside the fixture I believe.